sw/device/silicon_creator/lib/drivers/otbn.c

```diff
@@ -0,0 +1,113 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/drivers/otbn.h"
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/silicon_creator/lib/error.h"
+
+#include "otbn_regs.h"  // Generated.
+
+/**
+ * Ensures that `offset` and `size` are valid for a given `mem_size`.
+ *
+ * Valid are 32b word accesses to 32b-aligned memory locations within
+ * `mem_size`.
+ */
+static bool check_offset_len(uint32_t offset_bytes, size_t len_bytes,
+                             size_t mem_size) {
+  // The overflow check below assumes/requires two unsigned inputs.
+  return (len_bytes % sizeof(uint32_t) == 0 &&
+          offset_bytes % sizeof(uint32_t) == 0 &&
```

Comment:
I've added a TODO to switch this once #6112 is resolved.

Reply:
In general, I support migrating this to alignment helper functions.

Realistically, in this case, the modulo operation with a constant power-of-two divisor (ie: sizeof(uint32_t), aka 4) is going to get optimized by the compiler into a bitwise-AND with the value 3.

My concern in sec_mmio.c was because the divisor was not constant, but even then, my concern might have been misplaced.  `__udivdi3` is for long ints (aka uint64_t on rv32i, whereas in sec_mmio.c, the modulus is a 32/32 divide).

I'm not recommending a change here, just elaborating on my original reasoning in sec_mmio, and why my concern may have been misplaced.